packages/shared/src/react/contexts.tsx

```diff
@@ -50,6 +50,25 @@ const OrganizationProvider = ({
   );
 };
 
+const ClerkProviderAssertionContext = React.createContext<{ inTree: true } | undefined>(undefined);
+ClerkProviderAssertionContext.displayName = 'ClerkProviderAssertionContext';
+
+function useAssertWrappedByClerkProvider(displayNameOrCustomHandler?: string | (() => void)): void {
+  const assertionCtx = React.useContext(ClerkProviderAssertionContext);
+
+  if (!assertionCtx) {
+    if (typeof displayNameOrCustomHandler === 'function') {
+      displayNameOrCustomHandler();
+      return;
+    }
+
+    throw new Error(
+      `${displayNameOrCustomHandler || 'Clerk components'} must be wrapped within the <ClerkProvider> component.
+      Please see: https://clerk.com/docs/components/clerk-provider`,
```

Comment:
Similar suggestions as above.

Also, I didn't test run this but I removed the indentation on the second line, from my experience this would otherwise be indented.

```suggestion
      `${displayNameOrCustomHandler || 'Clerk components'} must be wrapped within the <ClerkProvider /> component.
Learn more: https://clerk.com/docs/components/clerk-provider`,
```